c++/src/kj/debug.h

```diff
@@ -283,14 +283,14 @@ namespace kj {
       KJ_UNIQUE_NAME(_kjContext)(KJ_UNIQUE_NAME(_kjContextFunc))
 
 #define KJ_REQUIRE_NONNULL(value, ...) \
-  (*({ \
+  (*[&] { \
```

Comment:
I worry that the IIFE syntax will be optimized differently from the `({...})` approach. Can we use it just for MSVC, and stick with the existing syntax for GCC/clang?

Reply:
Sure, done.